task/result.go

```diff
@@ -422,11 +422,23 @@ func (r *Result) ReportingRecord(record interface{}) *Result {
 	return r.addReportingRecord(record)
 }
 
+func CreateCodeComplianceRecord() CodeComplianceRecord {
+	record := ReportingRecord{
+		Type: "udm.CodeComplianceRecord",
+	}
+	return CodeComplianceRecord{
+		ReportingRecord: record,
+	}
+}
+
 // CreateDeploymentRecord creates a new reporting record for deployment.
 func CreateDeploymentRecord(client *http.HttpClient, taskInfo *DeploymentRecordTaskInfo, status DeploymentStatus) *DeploymentRecord {
-	record := &DeploymentRecord{
+	reporting := ReportingRecord{
```

Comment:
make it `reportingRecod`.

Reply:
It would be good to make naming change like Mayur suggested or just `record` at least or even better make it inline without introducing the variable since it's used only in the creation of `DeploymentRecord` no need for verbosity if it is obvious in my opinion.

Reply:
updated.

Reply:
umm this satisfies the form you want to have in JSON in the end but i am afraid that it is confusing. 

If we want it like this maybe we should think of adding a method on  ReportingRecord to assign a type and then have helper method that would enforce the correct constants for correct types. When I think more about it, I would remove this function here and add set of methods in new file where we would have complete set of reporting records (with ReportingRecord and other records definition) and add constants for the types. On Reporting Record I would make factory methods that would return correct objects with correctly set `Type`. 
This would improve original code pretty much and solve issue with confusing function name and "wrong" type here above. If you want I can contribute on this PR if it's too much of me to ask of you since it's kind of code pruning now.

Reply:
Talked in slack, for now removing the code compliance record code. @gpugar will take on this as a separate initiative to refactor reporting overall.